node/state.re

```diff
@@ -160,8 +150,10 @@ let apply_block = (state, block) => {
       },
     )
   );
-  switch (new_snapshot) {
-  | Some(new_snapshot) =>
+  if (block.state_root_hash == old_state.protocol.state_root_hash) {
+    Ok(state);
+  } else {
+    let new_snapshot = Protocol.hash(state.protocol);
```

Comment:
This may be problematic, before we hashed the state before the block application and now we're hashing it after, I'm not sure on the implications of this tho.